test-engineering/contract-tests/partner/app/record_keeper.py

```diff
@@ -0,0 +1,41 @@
+from multiprocessing.managers import SyncManager
+from typing import Dict, List
+
+from fastapi import Request
+
+from models import Header, Record, Records
+
+
+class RecordKeeper:
+    """Responsible for Contile request history management"""
+
+    def __init__(self, multi_process_manager: SyncManager) -> None:
+        """Create an instance of RecordKeeper."""
+
+        self._records: List[Record] = multi_process_manager.list()
+
+    def add(self, request: Request) -> None:
+        """Create record from Fast API Request and add record to the record keeper."""
+
+        headers: List[Header] = [
+            Header(name=header[0], value=header[1])
```

Comment:
[427b208](https://github.com/mozilla-services/contile/pull/415/commits/427b2085dbbcadd960e8f9f678df176585829577) - We cannot, see image:
<img width="524" alt="image" src="https://user-images.githubusercontent.com/3422688/178586575-3ad088de-c75a-47ef-86aa-9ff3a72c77a5.png">
However, as discussed, we can unpack the tuples to avoid further indexing.